QuestionBot/QuestionBot/Model/InMemoryStore.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+
+    public class InMemoryStore : IStore {
+        private IList<IRecord> allRecords = new List<IRecord>();
+        private static int _questionId = 0;
+
+        public IRecord CreateRecord(string question){
+            _questionId++;
+
+            Record newQuestion = new Record(_questionId, question, DateTime.Now);
+            allRecords.Add(newQuestion);
+
+            return newQuestion;
+        }
+
+        public bool TryUpdateRecord(int id, string answer){
+            IRecord recordToUpdate;
+            
+            try{
+                recordToUpdate = allRecords.First(recordItem => recordItem.ID == id);
```

Comment:
Is there a way we can do this without the try-catch? This code would work fine except that throwing and catching exceptions is expensive (performance-wise) so if we can easily avoid it then we should.